### PR TITLE
Changed text input into tags input in 'add collaborators' modal

### LINKED ISSUE #1909: Input box should not reload when it is empty
**Is your feature request related to a problem? Please describe.**
image
If we don't add anything in the input box and click on <b>add collaborators</b> button then the page reloads and give this error:
image

**Describe the solution you'd like**
We can enhance the user experience by providing a client side verification of the input box. If there is no value in the input box then we can prevent the reload and add border of red, so that the user knows that it is invalid.

**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by dipanshparmar**
Fixes #1909 

#### Describe the changes you have made in this PR -
1. Changed text input into tags input
2. Added functionality to convert pasted emails ids into tags
3. Tags can be added by pressing **space**, **comma** or **enter**.

### Screenshots of the changes (If any) -
**Initial State (Button is disabled)**
image

**On Input**
image

**On Paste (with new lines)**
newline

**On Paste (with spaces)**
spaces

### CODE REVIEW

**Comment by dipanshparmar** [comment]
@satu0king, @aayush-05, @Shivansh2407 inputs please.

File: app/views/projects/_add_collaborator_modal.html.erb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+<script src="https://cdn.jsdelivr.net/bootstrap.tagsinput/0.8.0/bootstrap-tagsinput.min.js"></script>
```
Comment (tachyons) [inline app/views/projects/_add_collaborator_modal.html.erb:1]: add this to packge.json instead

**Comment by nitin10s** [comment]
This feature actually enhances readability and is quite good. Great! 😄 One more addition, later on, can be - we can throw an error there and then when invalid emails are entered, so the user doesn't have to wonder which emails were invalid/valid.

File: app/views/projects/_add_collaborator_modal.html.erb (hunk 1 above)
Comment (dipanshparmar) [reply app/views/projects/_add_collaborator_modal.html.erb:1]: @tachyons I'm not sure how to add this in package.json

**Comment by dipanshparmar** [comment]
> This feature actually enhances readability and is quite good. Great! smile One more addition, later on, can be - we can throw an error there and then when invalid emails are entered, so the user doesn't have to wonder which emails were invalid/valid.

Sure @nitin10s, I'll create a new PR for this feature once this one gets approved. :slightly_smiling_face:

File: app/views/projects/_add_collaborator_modal.html.erb (hunk 1 above)
Comment (tachyons) [reply app/views/projects/_add_collaborator_modal.html.erb:1]: https://dev.to/mbackermann/how-and-when-to-use-yarn-on-rails-3jm4

Comment (dipanshparmar) [reply app/views/projects/_add_collaborator_modal.html.erb:1]: It seems like they don't provide any yarn version for this. Can you please confirm?

Comment (tachyons) [reply app/views/projects/_add_collaborator_modal.html.erb:1]: Seems so https://github.com/bootstrap-tagsinput/bootstrap-tagsinput/issues/553. So you can put the flle in `vendor` dir

Comment (tachyons) [reply app/views/projects/_add_collaborator_modal.html.erb:1]: On second thought it is not urgent, so merging this

**Comment by tachyons** [comment]
Thanks 👍

**Pull request merged**